gutils/gimagewritepng.c

```diff
@@ -149,7 +153,10 @@ return(false);
    png_write_info(png_ptr, info_ptr);
 
     if (color_type == PNG_COLOR_TYPE_RGB)
-	png_set_filler(png_ptr, '\0', PNG_FILLER_BEFORE);
+        png_set_filler(png_ptr, '\0', PNG_FILLER_AFTER);
```

Comment:
Don't know enough to judge about these.

Reply:
It's needed, see my comments in #3778. This is how I fixed my example "B" image from turning yellow. After I fixed that, I found another problem, where RGBA images would have their red and blue channels flipped, resulting in some pretty cool looking, but wrong, images, so telling libpng to treat it as BGR fixed that.

Reply:
By the way @skef, this was only ever used in `svg.c`&hellip;I doubt this was ever really stress-tested to the degree I've done with this commit (I made sure all my images in QRD.sfd, Chomsky.sfd and KJV1611.sfd worked, as well as making a file with all kinds of background images, monochrome, indexed, photographs, etc), because it was only used for PNG backgrounds in SVG&hellip;not something I ever used, and not something _useful_ to be frank.

My tests show `PNG_FILLER_BEFORE` is simply wrong.